vault/mount.go

```diff
@@ -350,19 +384,80 @@ func (c *Core) remount(src, dst string) error {
 	return nil
 }
 
+// tuneMount is used to set config on a mount point
+func (c *Core) tuneMount(path string, config MountConfig) error {
+	// Ensure we end the path in a slash
+	if !strings.HasSuffix(path, "/") {
+		path += "/"
+	}
+
+	// Prevent protected paths from being changed
+	for _, p := range protectedMounts {
+		if strings.HasPrefix(path, p) {
+			return fmt.Errorf("cannot tune '%s'", path)
+		}
+	}
+
+	// Verify exact match of the route
+	match := c.router.MatchingMount(path)
+	if match == "" || path != match {
+		return fmt.Errorf("no matching mount at '%s'", path)
+	}
+
+	// Find and modify mount
+	for _, ent := range c.mounts.Entries {
+		if ent.Path == path {
+			if config.MaxLeaseTTL != nil {
+				if *config.MaxLeaseTTL == 0 {
+					*ent.Config.MaxLeaseTTL = 0
+				} else {
+					ent.Config.MaxLeaseTTL = config.MaxLeaseTTL
+				}
+			}
+			if config.DefaultLeaseTTL != nil {
+				if *ent.Config.MaxLeaseTTL == 0 {
+					if *config.DefaultLeaseTTL > c.maxLeaseTTL {
+						return fmt.Errorf("Given default lease TTL of %d greater than system default lease TTL of %d",
+							*config.DefaultLeaseTTL, c.maxLeaseTTL)
+					}
+				} else {
+					if *ent.Config.MaxLeaseTTL != 0 && *ent.Config.MaxLeaseTTL < *config.DefaultLeaseTTL {
```

Comment:
I'm not sure what you mean here, but the Max logic has to come first, so that if the value is set in the call, that updated value is used when evaluating the Default logic.